node e2e: support creating gce instances from images in prep for pr builder

pwittrock opened the pull request:
@vishh

vishh commented on test/e2e_node/runner/run_e2e.go:
```diff
@@ -122,41 +137,102 @@ func main() {
+			{
+				AutoDelete: true,
+				Boot:       true,
+				Type:       "PERSISTENT",
```
Should we set the size?

pwittrock replied:
So far I have been leaving the defaults and they seem to have worked.  Do you think we should shrink the size from the default, or just for consistency?

vishh replied:
The default is `10GB` right? Would we ever use up all of that? May be disk usage tests?

pwittrock replied:
I doubt we would, but I also don't think we are saving much by shrinking it to save 6-8GB.  The CPU and ram are more costly.  The requested size cannot be smaller than the image size, so shrinking it would make creating images more error prone since we would have to always change the defaults.

pwittrock replied:
As a follow up, most base images (ubuntu, jessie,centos) are 10GB, and we can't shrink smaller than that.

vishh commented on test/e2e_node/runner/run_e2e.go:
```diff
@@ -122,41 +137,102 @@ func main() {
+		time.Sleep(time.Second * 30)
+	}
+	// Sleep an additional minute to let the vm be ready
+	time.Sleep(time.Minute)
```
Could this be flaky? Can we perform some health check instead like `[ -n ssh -- docker version | grep Server ]` or something similar?

pwittrock replied:
SGTM.

vishh commented on test/e2e_node/runner/run_e2e.go:
```diff
@@ -122,41 +137,102 @@ func main() {
+}
+
+func machineType() string {
+	return fmt.Sprintf("zones/%s/machineTypes/f1-micro", *zone)
```
Why micro? Are micro's good for performance tests? IIRC, they get throttled heavily..

pwittrock replied:
I didn't know what to pick, so just picked something cheap so as not to waste resources.  I am good with bumping this up.

pwittrock replied:
Done

timstclair commented on hack/verify-flags/known-flags.txt:
```diff
@@ -149,6 +149,7 @@ input-dirs
 insecure-bind-address
 insecure-port
 insecure-skip-tls-verify
+instance-name-prefix
```
Do you get an error without this? I would not expect verify-flags to include test flags...

vishh replied:
I believe it does.

pwittrock replied:
yeah you do :(

timstclair commented on test/e2e_node/gcloud/gcloud.go:
```diff
@@ -60,12 +61,16 @@ type CmdHandle struct {
 
 func (gc *gCloudClientImpl) Command(cmd string, moreargs ...string) ([]byte, error) {
-	args := append([]string{"compute", "ssh"})
+	args := []string{"compute"}
```
Does `--project` need to come before `ssh`?

pwittrock replied:
No, I think the ordering is gcloud <project flag> compute <zone flag> ssh

pwittrock replied:
Done

pwittrock commented:
just pushed something that will break this, will fix

pwittrock commented:
PTAL

timstclair commented:
LGTM, but you may want to wait for Vish's LGTM as well.

pwittrock commented:
@k8s-bot test node e2e experimental

pwittrock commented:
@k8s-bot test node e2e experimental

pwittrock commented:
@k8s-bot test node e2e experimental

pwittrock committed:
Node e2e test runner - run against images
- support allocating gce instances from images and running tests against them
- set --hostname-override to match node name
- add jenkins script to source to reproduce jenkins build locally

pwittrock commented:
@k8s-bot test node e2e experimental

pwittrock commented:
@k8s-bot test node e2e experimental

pwittrock commented:
@k8s-bot test node e2e experimental

pwittrock commented:
@k8s-bot test node e2e experimental

pwittrock commented:
@k8s-bot test node e2e experimental

pwittrock commented:
@k8s-bot test node e2e experimental

pwittrock commented:
@k8s-bot test node e2e experimental

pwittrock commented:
@k8s-bot test node e2e experimental

pwittrock commented:
PTAL
Passed on jenkins node pr builder http://kubekins.dls.corp.google.com:8081/job/node-pull-build-e2e-test/18/

pwittrock commented:
PTAL
See node e2e pr builder run 18

timstclair commented:
LGTM

pwittrock commented:
@k8s-bot test this issue: #21810
